qiita_db/metadata_template/base_metadata_template.py

```diff
@@ -513,7 +523,12 @@ def _clean_validate_template(cls, md_template, study_id,
         # In the database, all the column headers are lowercase
         md_template.columns = [c.lower() for c in md_template.columns]
 
-        # Droping/Ignoring internal generated colums
+        # Droping/Ignoring internal generated columns:
+        # A legacy feature of _clean_validate_template() is that it silently
+        # removes these columns before validating. It would be better to make
```

Comment:
Could you expand why will it be better to make it optional? Is this for testing? Do you think is worth adding an issue about this?

Reply:
I agree, making it optional might be a little much, if our intention is to always have it on. As you said, it is good for usability. My concern is that it be made explicitly known to the caller, because it is silent behavior that did catch me off guard; it might just be as simple as writing it into the method docs.

I would only make a separate issue out of it, if we think it might be better to move that usability behavior a level above this method, so that we can better write tests to this method.

Reply:
Good point, add as a Note in the docstring ...